transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/backends/BackendFactory.java

```diff
@@ -0,0 +1,20 @@
+// Copyright 2018 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime.backends;
+
+/** Used by backend discovery to create {@link TransportBackend}s. */
```

Comment:
can't we  use TransportBackend directly?

Reply:
Doing so will impose a requirement on the backend class to either:
1. have a certain static method e.g. like you suggested below `getInstance()`
2. have a constructor that takes certain arguments

which may not sound like a big deal, but it  violates the `single responsibility principle`, the TransportBackend is now responsible for communicating with the backend and for integrating with the discovery mechanism. Whereas having separate classes for this separates these concerns.